### ISSUE TITLE
Broken icons for installed webapp

### ISSUE THREAD

**Issue by atjn** [mastodon/mastodon#18735]
### Steps to reproduce the problem

1. Go to any mastodon instance in a browser such as Firefox or Chrome on an Android phone
2. Install the site through the browser as a webapp (sometimes called "add to home screen")
3. Go to the home screen where the app has been added
4. Observe that the logo/icon for the app has some weird edges around it:

Firefox on Android:
![image](https://user-images.githubusercontent.com/20535393/173784641-349a3b62-d8dd-4dba-b851-c46a482ec26e.png)

Chrome on Android:
![image](https://user-images.githubusercontent.com/20535393/173784854-bf5c8c4f-e5a6-406b-b74e-83c88524f8c9.png)

This happens because Mastodon doesn't provide `maskable` icons. Here is the warning from a Lighthouse test:
![image](https://user-images.githubusercontent.com/20535393/176032704-aa771e49-a4dd-41bf-a835-5b3b14aa7810.png)
[Lean more](https://web.dev/maskable-icon-audit)

The maskable icons would need to be added to the [`manifest_serializer`](https://github.com/mastodon/mastodon/blob/main/app/serializers/manifest_serializer.rb) to fix this issue.

This issue is not observed on an iPhone, but that is only because you provide a proprietary `apple-touch-icon`, which has the same function as the `maskable` icons, but exclusively for Safari on iOS. Newer versions of Safari do not require this proprietary icon, and can instead accept `maskable` icons. 

Related: #18115

### Expected behaviour

The app icon should have a solid background color that extends to the edges of the icon, like most other apps and webapps have.

### Actual behaviour

The app icon has random consistency issues where bars of other colors are displayed at the edges of the icon.

### Specifications

Any recent version of Mastodon, tested up to version 3.5.3

Any recent version of Android (certainly Android 11 and 12, maybe earlier). Any recent version of a browser that supports "Install" or "Add to home screen" functionality ([see versions here](https://caniuse.com/web-app-manifest), certainly Chrome 100+ and Firefox 100+).

This is theoretically also an issue outside of Android, although it is by far the largest platform to be affected today. The `maskable` icons are designed as a standard that is usable by any OS.

**Comment by danger89** [comment]
I also see this 404 not found problems in the logging of my Nginx server (the backend so to say):
```log
xxx.xxx.xxx.xx - - [23/Nov/2022:00:02:02 +0100]  "GET /apple-touch-icon.png HTTP/2.0" 404 650 "-" "*SECRET*" "mastodon.melroy.org" 0.005
```


The `apple-touch-icon.png` and other icons for Apple products should be generated and place in the root folder of the Mastodon server. 

I'm asking Mastodon devs to add those missing files. Avoiding those 404's.

**Comment by bici** [comment]
 no longer works on mastodon

**Comment by atjn** [comment]
This is fixed by #20904, altough I think it could be improved, see https://github.com/mastodon/mastodon/pull/20904#issuecomment-1893503544
